### PR TITLE
Msal-Broker interaction changes

### PR DESCRIPTION
**Pull request by kreedula**

### CODE REVIEW

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        IMicrosoftAuthService service;
+
+        final MicrosoftAuthClient client = new MicrosoftAuthClient(request.getAppContext());
+        final MicrosoftAuthServiceFuture authServiceFuture = client.connect();
+
+        try {
+            service = authServiceFuture.get();
+            interactiveRequestIntent = service.getIntentForInteractiveRequest();
+
+            // TODO : See what's the right exception here
```
Comment (iambmelt) [inline msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java]: My assumption would be that these errors roll-up to `ClientException` unless a good case can be made for any other error

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerActivity.java [hunk 2]
```diff
@@ -89,8 +89,18 @@ protected void onSaveInstanceState(final Bundle outState) {
...
+        final String methodName = ":onActivityResult";
+        Logger.info(TAG + methodName,
+                "Result received from Broker "
+                        + "Request code: " + requestCode
+                        +  " Result code: " + requestCode
+        );
+
+        if (resultCode == AuthenticationConstants.UIResponse.TOKEN_BROKER_RESPONSE ||
+                resultCode == AuthenticationConstants.UIResponse.BROWSER_CODE_CANCEL
+                || requestCode == AuthenticationConstants.UIResponse.BROWSER_CODE_ERROR) {
```
Comment (heidijinxujia) [inline msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerActivity.java]: requestCode  ->resultCode ?

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    @Override
+    public AcquireTokenResult acquireTokenSilent(AcquireTokenSilentOperationParameters parameters) throws BaseException {
+        IMicrosoftAuthService service;
+
+        MicrosoftAuthClient client = new MicrosoftAuthClient(parameters.getAppContext());
+        MicrosoftAuthServiceFuture future = client.connect();
+
+        try {
+            //Do we want a time out here?
+            service = future.get();
```
Comment (heidijinxujia) [inline msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java]: Yeap, and we could call MicrosoftAuthServiceFuture.get(long l, TimeUnit timeUnit) instead to set the timeout.

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/MSALControllerFactory.java [hunk 4]
```diff
@@ -88,10 +88,11 @@ public static BaseController getAcquireTokenController(@NonNull final Context ap
                                                                         @NonNull final Authority authority,
                                                                         @NonNull final PublicClientApplicationConfiguration applicationConfiguration) {
         List<BaseController> controllers = new ArrayList<>();
-        controllers.add(new LocalMSALController());
         if (brokerEligible(applicationContext, authority, applicationConfiguration)) {
-            controllers.add(new BrokerMSALController());
+            controllers.add(new BrokerMsalController());
         }
+        controllers.add(new LocalMSALController());
```
Comment (heidijinxujia) [inline msal/src/main/java/com/microsoft/identity/client/internal/controllers/MSALControllerFactory.java]: Why does it need to add both lcoal msal controller as well when the broker msal controller is already added?

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java (hunk 1 above)
Comment (kreedula) [reply msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java]: Agreed, missed to resolve todo's , fixed them

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java (hunk 3 above)
Comment (kreedula) [reply msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java]: Yes the question is what's the time limit though, I will create another work item to discuss and add this we needed

File: msal/src/main/java/com/microsoft/identity/client/internal/controllers/MSALControllerFactory.java (hunk 4 above)
Comment (kreedula) [reply msal/src/main/java/com/microsoft/identity/client/internal/controllers/MSALControllerFactory.java]: Reverted the change back. The idea behind the design is we look in local msal first and then broker

**Pull request merged**